pageserver/src/tenant/timeline.rs

```diff
@@ -324,7 +334,7 @@ pub struct Timeline {
     pub walreceiver: Mutex<Option<WalReceiver>>,
 
     /// Relation size cache
-    pub rel_size_cache: RwLock<HashMap<RelTag, (Lsn, BlockNumber)>>,
+    pub(crate) rel_size_cache: RwLock<RelSizeCache>,
```

Comment:
I wasn't aware of the existence of this cache and I'm just noticing that its size grows linearly with the number of relations (albeit not by much). I'd really like it if we would bound this to a certain number of relations (can be rather large). A simple way to deal with this is to evict say 1/4 of the entries when the size threshold is reached.

I know it was unbounded before this patch, but this feels like a good opportunity to address it.

Reply:
I'm reluctant to expand the scope of this PR, because this is a blocker for PR #7377, which in turn is somewhat urgent. It's a non-trivial amount of work to figure out the eviction policy, how large the cache should be etc. I agree that in the long run a fixed size cache is not great, and we should do that eventually.

Reply:
Makes sense. Would be great if you could open an issue for the unbounded cache. Feel free to resolve.

Reply:
Created https://github.com/neondatabase/neon/issues/7450